test/no-function-overriding.js

```diff
@@ -0,0 +1,137 @@
+/**
+ * @fileoverview Test for no-function-overriding rule
+ * @author Nicolas Feignon <nfeignon@gmail.com>
+ */
+
+'use strict';
+
+var Solium = require ('solium');
+
+var userConfig = {
+    rules: {
+        "security/no-function-overriding": "error"
+    }
+};
+
+describe ('[RULE] no-function-overriding: Acceptances', function () {
+
+    it ('should accept contracts that don\'t override their inherited functions', function (done) {
+        var code = [
+            `
+                contract A {
+                    function foo();
+                    function ();
+                }
+                contract B is A {
+                    function bar();
+                    function ();
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function bar();
+                    function();
+                }
+                contract C {
+                    function foo();
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function bar();
+                    function foo(uint);
+                }
+            `,
+            `
+                contract A {
+                    function foo();
+                }
+                contract B is A {
+                    function foo(uint, string);
+                }
+                contract C is B {
+                    function foo(uint);
+                }
+            `
+        ];
+        var errors;
+
+        for (let expr of code) {
+            errors = Solium.lint (expr, userConfig);
+            errors.length.should.equal (0);
+        }
+
+        Solium.reset ();
+        done ();
+    });
+});
+
+describe ('[RULE] no-function-overriding: Rejections', function () {
+
+    it ('should reject contracts that override their inherited functions', function (done) {
+        var code = [
```

Comment:
Is below eg considered overriding:
```
contract A {
    function foo();
}

contract B is A {
    function foo() {
        require(true);
    }
}
```

`foo` is abstract in A but defined in B

I think this too is overriding by definition. Please confirm it and if yes, then ensure that the rule can handle this case and add tests for it.

Reply:
It is indeed overriding. See my response with the quote to Solidity doc above. I also tested in Remix. I'll add the test.